src/luaState.cpp

```diff
@@ -186,18 +313,77 @@ Variant LuaState::callFunction(String functionName, Array args) {
 	return toReturn;
 }
 
+Variant LuaState::callFunctionRegistry(String functionName, Array args) {
```

Comment:
The main reason we keep the call function and function exists methods for the main lua state is to maintain comparability, but I think with the registry in mind it would make a little more sense to just push and pull callable. I know there is a memory leak related to that currently, but hopefully that will be resolved soon. It would for sure be resolved before 2.1 is stable. So I do not want to make design choices to work around that quite yet.

Reply:
Calling functions in the registry is currently setting a work-around for a well-known issue of what you just said.
If the question is "Why the registry" the answer is that for sandboxing lua within lua, you treat the global environment as dirty, as it can always be altered.

Reply:
"Calling functions in the registry is currently setting a work-around for a well-known issue"

>I know there is a memory leak related to that currently, but hopefully that will be resolved soon. It would for sure be resolved before 2.1 is stable. So I do not want to make design choices to work around that quite yet.

Like I said, I do not want to make design choices around this, as now it is a method we need to support.

Reply:
I am saying, i know this works around the memory leak, but I do not want to implement such a work around as it will make the API more dirty in the long run and that issue will be resolved before the current version is stable.

Reply:
Ideally LuaAPI.call_function will be retired as well, I'm more hesitant to get rid of that one though as its been around for awhile.

Reply:
> Ideally LuaAPI.call_function will be retired as well, I'm more hesitant to get rid of that one though as its been around for awhile.

.... create a LuaAPI._unref function?

Reply:
> .... create a LuaAPI._unref function then immedieately mark it as deprecated?

Like I said, my goal is not to work around this bug. v2.1 will not be released before this bug is fixed.

Reply:
> > .... create a LuaAPI._unref function then immedieately mark it as deprecated?
> 
> Like I said, my goal is not to work around this bug. v2.1 will not be released before this bug is fixed.

Well... i have no idea what to do.

Reply:
> Well... i have no idea what to do.

The TLDR is I do not want separate methods to call functions or check if they exists, they exists on the main LuaAPI object for comparability reasons, but moving forward I want to lean more into just having methods be returned as a Callable. I also do not want to introduce more methods to work around a memory leak that will be resolved before the next stable release. So all that needs to be done is `call_function_registry` and `registry_function_exists` should be removed.

Reply:
> The TLDR is I do not want separate methods to call functions or check if they exists, they exists on the main LuaAPI object for comparability reasons, but moving forward I want to lean more into just having methods be returned as a Callable. I also do not want to introduce more methods to work around a memory leak that will be resolved before the next stable release. So all that needs to be done is `call_function_registry` and `registry_function_exists` should be removed.

is `call_function` and `function_exists` be marked as deprecated later then?

Reply:
> is `call_function` and `function_exists` be marked as deprecated later then?

Yes, thats more of a question of when, that probably wont be until the next major version bump. The less codebase changes that need to happen for the end users between minor version the better, ideally it will be 0. And since those have been in the API since v1 i am hesitant to get rid of them.